<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Michael C Turchin" />

<meta name="date" content="2019-05-16" />

<title>bmass Introduction – Simulated Data</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' || rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">bmass Introduction – Simulated Data</h1>
<h4 class="author">Michael C Turchin</h4>
<h4 class="date">2019-05-16</h4>



<p>This is a very simple, introductory vignette on how to run <code>bmass</code>. In this introductory example, we will use a small, simulated dataset to run <code>bmass</code> and explore some basic output. The dataset is provided with the package.</p>
<p>The purpose of this vignette is to quickly allow users to get up and running with <code>bmass</code>. For a more advanced introductory vignette, where we download and use the GlobalLipids 2013 dataset as an example, please see <a href="http://mturchin20.github.io/bmass/articles/bmassIntro.2.RealData.html">here</a>.</p>
<div id="running-bmass" class="section level2">
<h2>Running <code>bmass</code></h2>
<p>For this introductory vignette, we will be using <code>bmass_SimulatedData1</code>, <code>bmass_SimulatedData2</code>, and <code>bmass_SimulatedSigSNPs</code>. These are simulated datasets created for the purposes of unit testing and this vignette. To load up ‘bmass’ and the datasets, run the following:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">library</span>(<span class="st">&quot;bmass&quot;</span>);</a>
<a class="sourceLine" id="cb1-2" title="2"><span class="kw">data</span>(bmass_SimulatedData1, bmass_SimulatedData2, bmass_SimulatedSigSNPs)</a>
<a class="sourceLine" id="cb1-3" title="3">Phenotypes &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;bmass_SimulatedData1&quot;</span>, <span class="st">&quot;bmass_SimulatedData2&quot;</span>)</a></code></pre></div>
<p>First, we’ll take a look at the format of one of our simulated datasets:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">head</span>(bmass_SimulatedData1)</a></code></pre></div>
<pre><code>##   Chr    BP Marker  MAF A1 A2 Direction  pValue    N
## 1   1  1000    rs1 0.20  A  G         - 6.0e-13 2500
## 2   1  2000    rs2 0.10  G  T         - 7.6e-02 2467
## 3   2  3000    rs3 0.06  T  G         + 3.0e-09 2761
## 4   3  4000    rs4 0.40  C  A         + 5.6e-03 2310
## 5   3 15000    rs5 0.40  C  T         + 1.0e-07 2410
## 6   3 21000    rs6 0.37  G  A         + 5.0e-08 2582</code></pre>
<p>Here you should see nine columns: Chr, BP, Marker, MAF, A1, A2, Direction, pValue, N. <code>bmass</code> expects each input dataset (representing a single phenotype GWAS) to contain these 9 columns and with these specific headers. Their descriptions are as follows –</p>
<ul>
<li>Chr: Chromosome</li>
<li>BP: Basepair Position</li>
<li>Marker: Variant ID information, such as a rsID # (should be a unique value for each variant)</li>
<li>MAF: Minor Allele Frequency</li>
<li>A1: Reference allele</li>
<li>A2: AlternativeaAllele</li>
<li>Direction: Direction of effect size (a column of “+” or “-”)</li>
<li>pValue: p-Value from univariate GWAS</li>
<li>N: Sample size</li>
</ul>
<p>Now to run <code>bmass</code>, all we use is the following code:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1">bmassResults &lt;-<span class="st"> </span><span class="kw">bmass</span>(Phenotypes, bmass_SimulatedSigSNPs);</a></code></pre></div>
<p>Note that at its most basic level, we only need to provide two inputs for <code>bmass()</code> to properly run: a vector of strings containing the variable names of each of our univariate summary GWAS datasets and a file containing the list of previously associated univariate genome-wide significant SNPs. The former input, the vector of strings, dictates to <code>bmass()</code> where to find each dataset and what to call each phenotype being analyzed (ie the name of each variable containg the GWAS summary data is expected to correspond to the phenotype represented by that data). The latter input is just a list of each previous GWAS SNP with the following two columns of information, <code>Chr</code> and <code>BP</code>:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">bmass_SimulatedSigSNPs</a></code></pre></div>
<pre><code>##   Chr   BP
## 1   1 1000
## 2   2 3000</code></pre>
</div>
<div id="exploring-basic-results" class="section level2">
<h2>Exploring basic results</h2>
<p>Now to begin to get a sense of what <code>bmass()</code> outputs, we can run the following:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"><span class="kw">summary</span>(bmassResults)</a></code></pre></div>
<pre><code>##                       Length Class  Mode     
## MarginalSNPs            3    -none- list     
## PreviousSNPs            4    -none- list     
## NewSNPs                 3    -none- list     
## LogFile                19    -none- character
## ZScoresCorMatrix        4    -none- numeric  
## Models                 18    -none- numeric  
## ModelPriors           126    -none- numeric  
## GWASlogBFMinThreshold   1    -none- numeric</code></pre>
<p>As you can see, there are multiple output variables corresponding to different components of the results. For the purposes of this vignette, we will only touch on the first three variables shown from this <code>summary()</code> output: <code>PreviousSNPs</code>, <code>MarginalSNPS</code>, and <code>NewSNPs</code>. <code>PreviousSNPs</code> refers to information on the input GWAS SNPs we used for the analysis, <code>MarginalSNPs</code> refers to information on the the marginally significant SNPs that were used in the final steps of the analysis, and <code>NewSNPs</code> refers to information on any SNPs that reach multivariate genome-wide levels of significance.</p>
<p>The main result most users will be immediately interested in is whether there are any new variants identified as genome-wide multivariate significant, and this can be determined by accessing the <code>NewSNPs</code> sublist:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1"><span class="kw">summary</span>(bmassResults<span class="op">$</span>NewSNPs)</a></code></pre></div>
<pre><code>##            Length Class      Mode   
## SNPs       22     data.frame list   
## logBFs     27     -none-     numeric
## Posteriors 27     -none-     numeric</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1"><span class="kw">head</span>(bmassResults<span class="op">$</span>NewSNPs<span class="op">$</span>SNPs)</a></code></pre></div>
<pre><code>##    ChrBP Chr   BP Marker  MAF A1 bmass_SimulatedData1_A2
## 9 4_7000   4 7000    rs8 0.15  G                       C
##   bmass_SimulatedData1_Direction bmass_SimulatedData1_pValue
## 9                              +                       7e-08
##   bmass_SimulatedData1_N bmass_SimulatedData1_ZScore
## 9                   2514                    5.391171
##   bmass_SimulatedData2_Direction bmass_SimulatedData2_pValue
## 9                              -                       6e-08
##   bmass_SimulatedData2_N bmass_SimulatedData2_ZScore GWASannot   mvstat
## 9                   2514                   -5.418801         0 145.0667
##   mvstat_log10pVal  unistat unistat_log10pVal Nmin logBFWeightedAvg
## 9         31.50084 29.36341          7.221849 2514           29.107</code></pre>
<p>As you can see in this example, we have 1 new SNP that reaches multivariate genome-wide significance. This is determined by using the smallest log BF weighted average value among the previous univariate significanct GWAS SNPs (<code>PreviousSNPS</code>):</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1">bmassResults<span class="op">$</span>GWASlogBFMinThreshold</a></code></pre></div>
<pre><code>## [1] 7.41173</code></pre>
<p>Any <code>MarginalSNPs</code> with <code>logBFWeightedAvg</code> greater than this value are designated as new multivariate associations (see eq. 1 in <a href="https://www.biorxiv.org/">Turchin and Stephens bioRxiv 2019</a>).</p>
<p>For further exploration and details of <code>bmass</code> output, please see the second introductory vignette.</p>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
